src/docgen/schema.ts

```diff
@@ -0,0 +1,335 @@
+/**
+ * Describes a type.
+ */
+export interface TypeSchema {
+
+  /**
+   * The type FQN. If missing, the type is a reference (array, map..)
+   */
+  readonly fqn?: string;
+
+  /**
+   * The name of the type (Map, List...)
+   */
+  readonly name: string;
+
+  /**
+   * The various types of the reference.
+   */
+  readonly types?: TypeSchema[];
```

Comment:
Is this for union types?

Reply:
this is for child types, aka a list of lambda.Code objects in Python appears as
```ts
{
  name: `typing.List[%]`,
  types: [{
    fqn: '@aws-cdk/aws-lambda.Code',
    name: 'aws_lambda.Code',
    types: [],
  }],
}
```
The `%` to represent how to place child types is something I would prefer the consumer to dictate, and was experimenting with allowing the passing of a function similar to `linkFormatter` in the MD. We could change `name` to be `string[]` and pass `['typing.List[', ']']` or something, but basically what clients need is access to the transliterated name for types and fqn, with some indication of how to layout any nested types, but just a giant string isn't good enough cause you need to know where all the discrete types are so you can build appropriate links to named types that may be a subset of the overall signature string that is displayed.

@rix0rrr we discussed this briefly at one point